build/repo.targets

```diff
@@ -1,50 +1,57 @@
 
-    <Scenarios Include="-n Json --kestrelTransport Sockets" />
-    <Scenarios Include="-n Json --webHost HttpSys" />
-    <Scenarios Include="-n Json -f Benchmarks.PassthroughConnectionFilter" />
+    <PhysicalScenarios Include="-n Json --kestrelTransport Sockets" />
+    <PhysicalScenarios Include="-n Json --webHost HttpSys" />
+    <PhysicalScenarios Include="-n Json -f Benchmarks.PassthroughConnectionFilter" />
 
-    <Scenarios Include="-n Json -m https" />
-    <Scenarios Include="-n Json -m https --kestrelTransport Sockets" />
-    <Scenarios Include="-n Json -m https --webHost HttpSys" />
+    <PhysicalScenarios Include="-n Json -m https" />
+    <PhysicalScenarios Include="-n Json -m https --kestrelTransport Sockets" />
+    <PhysicalScenarios Include="-n Json -m https --webHost HttpSys" />
 
-    <Scenarios Include="-n StaticFiles --path plaintext" />
+    <PhysicalScenarios Include="-n StaticFiles --path plaintext" />
 
-    <Scenarios Include="-n MvcPlaintext" />
+    <PhysicalScenarios Include="-n MvcPlaintext" />
 
-    <Scenarios Include="-n MvcJson" />
+    <PhysicalScenarios Include="-n MvcJson" />
 
-    <Scenarios Include="-n MemoryCachePlaintext" />
-    <Scenarios Include="-n MemoryCachePlaintextSetRemove" />
+    <PhysicalScenarios Include="-n MemoryCachePlaintext" />
+    <PhysicalScenarios Include="-n MemoryCachePlaintextSetRemove" />
 
-    <Scenarios Include="-n ResponseCachingPlaintextCached" />
-    <Scenarios Include="-n ResponseCachingPlaintextCached --method DELETE" />
-    <Scenarios Include="-n ResponseCachingPlaintextResponseNoCache" />
-    <Scenarios Include="-n ResponseCachingPlaintextRequestNoCache" />
-    <Scenarios Include="-n ResponseCachingPlaintextVaryByCached" />
+    <PhysicalScenarios Include="-n ResponseCachingPlaintextCached" />
+    <PhysicalScenarios Include="-n ResponseCachingPlaintextCached --method DELETE" />
+    <PhysicalScenarios Include="-n ResponseCachingPlaintextResponseNoCache" />
+    <PhysicalScenarios Include="-n ResponseCachingPlaintextRequestNoCache" />
+    <PhysicalScenarios Include="-n ResponseCachingPlaintextVaryByCached" />
   </ItemGroup>
 
   <PropertyGroup>
     <BenchmarksDependsOn Condition="'$(NoBuild)' != 'true'">Compile</BenchmarksDependsOn>
   </PropertyGroup>
 
   <Target Name="TestBenchmarks" DependsOnTargets="$(BenchmarksDependsOn)">
-    <Error Text="BENCHMARK_SERVER must be set" Condition="'$(BENCHMARK_SERVER)' == ''" />
-    <Error Text="BENCHMARK_CLIENT must be set" Condition="'$(BENCHMARK_CLIENT)' == ''" />
+    <Error Text="BENCHMARK_SERVER_PHYSICAL must be set" Condition="'$(BENCHMARK_SERVER_PHYSICAL)' == ''" />
+    <Error Text="BENCHMARK_CLIENT_PHYSICAL must be set" Condition="'$(BENCHMARK_CLIENT_PHYSICAL)' == ''" />
+    <Error Text="BENCHMARK_SERVER_CLOUD must be set" Condition="'$(BENCHMARK_SERVER_CLOUD)' == ''" />
+    <Error Text="BENCHMARK_CLIENT_CLOUD must be set" Condition="'$(BENCHMARK_CLIENT_CLOUD)' == ''" />
```

Comment:
Should we make one or both run types conditional? E.g. fail if none of the cloud or physical variables are set or if (say) the cloud server variable is set but not the cloud client variable. Then, the `<MsBuild>` tasks below would be conditional on the necessary variables.
The alternative is to assume anyone subsetting the tests would edit this file temporarily (including removing `<Error>` tasks as needed.

Reply:
This target is mainly used by the CI where both will be set.  I don't want to put more effort into refactoring this file unless people actually use it for ad-hoc testing.   @natemcmaster said he does occasionally, but I think most people run `BenchmarksDriver` manually for ad-hoc testing.

Reply:
@dougbu: Based on my own ad-hoc usage, I've changed my mind and updated the PR.  I think it's sufficient to make the tasks conditional on the env vars without any error messages.

Reply:
👍 Looks good and matches my ad-hoc usage as well.